gobblin-api/src/main/java/gobblin/writer/StreamEncoder.java

```diff
@@ -0,0 +1,25 @@
+package gobblin.writer;
+
+import java.io.OutputStream;
+
+import gobblin.annotation.Alpha;
+
+
+/**
+ * Interface for an object that can encrypt a bytestream.
+ */
+@Alpha
+public interface StreamEncoder {
```

Comment:
The advantage of having them together is that it is very obvious, for each encoding, how to decode it. Even if the names are similar, I could see someone spending too much time figuring that out.